File: demo_avanzado_transpilado.py
import json
import os as sistema_operativo

import json
import os as sistema_operativo
# desde math importar sqrt
@property
@staticmethod
class Animal:
    def __init__(self):
        self.nombre = None
        self.edad = None
    def hacer_sonido(self):
        print("El animal hace un sonido")


class Perro(Animal):
    def __init__(self):
        self.raza = None
    def hacer_sonido(self):
        print("¡Guau! Soy " + self.nombre)

    def info_especie(self):
        print("Los perros son mamíferos")


@staticmethod
async def obtener_datos(url):
    try:
        resultado = esperar fetch(url)
        datos = esperar resultado.json()
        return datos
    except NetworkError as error:
        print("Error de red: " + str(error))
        return None
    finally:
        print("Operación completada")

def saludar(nombre="Invitado", idioma="es"):
    if idioma == "es":
        print("¡Hola, " + nombre + "!")
    else:
        print("Hello, " + nombre + "!")

configuracion = {"debug": True, "version": "1.0"}
coordenadas = (10, 20, 30)
numeros_unicos = {1, 2, 3, 4, 5}
nombres = ["Ana", "Carlos", "María"]
configuracion["puerto"] = 8080
configuracion["host"] = "localhost"
numeros_unicos.add(6)
numeros_unicos.add(7)
cuadrados = [x*x for x en range(1, 6) if x % 2 == 0]
with open("datos.txt", "w") as archivo:
    # archivo.write("Datos de prueba")
for i in range(0, 5):
    print("Número: " + str(i))
for nombre in nombres:
    if nombre.startswith("A"):
        print("Nombre que empieza con A: " + nombre)
while contador < 3:
    print("Contador: " + str(contador))
def numeros_fibonacci():
    a, b = 0, 1
    yield a
    a, b = b, a + b

cuadrado = lambda x: x * x
suma = lambda a, b: a + b
try:
    resultado = 10 / 0
except ZeroDivisionError as error:
    print("No se puede dividir por cero")
except Exception as error:
    print("Error inesperado: " + str(error))
finally:
    print("Limpieza completada")
assert len(nombres) > 0, "La lista no puede estar vacía"
global contador_global
def main():
mi_perro = Perro()
    mi_perro.nombre = "Rex"
    mi_perro.edad = 3
    mi_perro.raza = "Labrador"
    mi_perro.hacer_sonido()
    Perro.info_especie()
    saludar("María", "es")
    saludar()
    print("Configuración: " + str(configuracion))
    print("Coordenadas: " + str(coordenadas))
    print("Números únicos: " + str(numeros_unicos))
    print("Cuadrados pares: " + str(cuadrados))
    print("Cuadrado de 5: " + str(cuadrado(5)))
    print("Suma de 3 y 7: " + str(suma(3, 7)))
    print("Demo completado exitosamente")

main()

File: demo_avanzado_transpilado.js
import json;
import * as sistema_operativo from os;

import json;
import * as sistema_operativo from os;
// desde math importar sqrt
// @property
// @staticmethod
class Animal {
  constructor() {
    this.nombre = null;
    this.edad = null;
}
  hacer_sonido() {
    console.log("El animal hace un sonido");
  }
}
class Perro extends Animal {
  constructor() {
    this.raza = null;
}
  hacer_sonido() {
    console.log("¡Guau! Soy " + this.nombre);
  }
  info_especie() {
    console.log("Los perros son mamíferos");
  }
}
// @staticmethod
async function obtener_datos(url) {
  try {
    resultado = esperar fetch(url);
    datos = esperar resultado.json();
    return datos;
  } catch (error) { // NetworkError
    console.log("Error de red: " + String(error));
    return None;
  } finally {
    console.log("Operación completada");
  }
}
function saludar(nombre = "Invitado", idioma = "es") {
  if (idioma == "es") {
    console.log("¡Hola, " + nombre + "!");
  } else {
    console.log("Hello, " + nombre + "!");
  }
}
let configuracion = {"debug": True, "version": "1.0"};
tupla coordenadas = (10, 20, 30);
let numeros_unicos = {1, 2, 3, 4, 5};
let nombres = ["Ana", "Carlos", "María"];
configuracion["puerto"] = 8080;
configuracion["host"] = "localhost";
numeros_unicos.add(6);
numeros_unicos.add(7);
comprension lista cuadrados = x*x para x en range(1, 6) si x % 2 == 0;
// con open("datos.txt", "w") como archivo
// archivo.write("Datos de prueba")
// fin con
for (let i = 0; i < 5; i++) {
  console.log("Número: " + String(i));
}
for (let nombre of nombres) {
  if (nombre.startswith("A")) {
    console.log("Nombre que empieza con A: " + nombre);
  }
}
contador = 0;
while (contador < 3) {
  console.log("Contador: " + String(contador));
  contador = contador + 1;
}
function numeros_fibonacci() {
  a, b = 0, 1;
  yield a;
  a, b = b, a + b;
}
}
cuadrado = lambda x: x * x;
suma = lambda a, b: a + b;
try {
resultado = 10 / 0;
} catch (error) { // ZeroDivisionError
console.log("No se puede dividir por cero");
} catch (error) {
console.log("Error inesperado: " + String(error));
} finally {
console.log("Limpieza completada");
}
// afirmar len(nombres) > 0 mensaje "La lista no puede estar vacía"
// global contador_global
contador_global = 100;
function main() {
let mi_perro = new Perro();
mi_perro.nombre = "Rex";
mi_perro.edad = 3;
mi_perro.raza = "Labrador";
mi_perro.hacer_sonido();
Perro.info_especie();
saludar("María", "es");
saludar();
console.log("Configuración: " + String(configuracion));
console.log("Coordenadas: " + String(coordenadas));
console.log("Números únicos: " + String(numeros_unicos));
console.log("Cuadrados pares: " + String(cuadrados));
console.log("Cuadrado de 5: " + String(cuadrado(5)));
console.log("Suma de 3 y 7: " + String(suma(3, 7)));
console.log("Demo completado exitosamente");
}
main();